supabase_auth/helpers.py

```diff
@@ -192,15 +194,46 @@ def handle_exception(exception: Exception) -> AuthError:
         return AuthUnknownError(get_error_message(error), e)
 
 
-def decode_jwt_payload(token: str) -> Any:
-    parts = token.split(".")
-    if len(parts) != 3:
-        raise ValueError("JWT is not valid: not a JWT structure")
-    base64url = parts[1]
+def str_from_base64url(base64url: str) -> str:
     # Addding padding otherwise the following error happens:
     # binascii.Error: Incorrect padding
     base64url_with_padding = base64url + "=" * (-len(base64url) % 4)
-    return loads(urlsafe_b64decode(base64url_with_padding).decode("utf-8"))
+    return urlsafe_b64decode(base64url_with_padding).decode("utf-8")
+
+
+def base64url_to_bytes(base64url: str) -> bytes:
+    # Addding padding otherwise the following error happens:
+    # binascii.Error: Incorrect padding
+    base64url_with_padding = base64url + "=" * (-len(base64url) % 4)
+    return urlsafe_b64decode(base64url_with_padding)
+
+
+class DecodedJWT(TypedDict):
+    header: JWTHeader
+    payload: JWTPayload
+    signature: bytes
+    raw: Dict[str, str]
+
+
+def decode_jwt(token: str) -> DecodedJWT:
+    parts = token.split(".")
+    if len(parts) != 3:
+        raise AuthInvalidJwtError("Invalid JWT structure")
+
+    # regex check for base64url
+    # for part in parts:
+    #     if not re.match(BASE64URL_REGEX, part):
+    #         raise AuthInvalidJwtError("JWT not in base64url format")
+
```

Comment:
This is failing with valid tokens, need to verify why.

Reply:
Is it still?

Reply:
fixed, it was missing the `IGNORECASE` flag 🫣